src/elements/text-element.js

```diff
@@ -59,4 +59,30 @@ export default class TextElement extends TextualElement {
   get hasPattern() {
     return this.pattern && this.pattern.length;
   }
+
+  get hasPatternDescription() {
+    return this.patternDescription && this.patternDescription.length;
+  }
+
+  toJSON() {
+    const json = super.toJSON();
+
+    json.numeric = this._isNumeric;
+
+    if (this.isNumeric) {
+      json.format = this.format || 'decimal';
+      json.min = this.hasMin ? this.min : null;
+      json.max = this.hasMax ? this.max : null;
+    } else {
+      json.pattern = this.hasPattern ? this.pattern : null;
+      json.pattern_description = this.hasPatternDescription ? this.patternDescription : null;
+    }
+
+    json.min_length = this.minLength;
```

Comment:
This should probably be `this._minLength` so it doesn't use the override runtime value. The "override" values are value set at runtime from data events via `SETMINLENGTH()` which should only impact runtime behavior and not the element serialization.

Reply:
Makes sense. Done.